neuralpredictors/measures/np_functions.py

```diff
@@ -12,10 +19,148 @@ def corr(y1, y2, axis=-1, eps=1e-8, **kwargs):
         y2:      second NumPy array
         axis:    dimension(s) along which the correlation is computed. Any valid NumPy axis spec works here
         eps:     offset to the standard deviation to avoid exploding the correlation due to small division (default 1e-8)
-        **kwargs: passed to final numpy.mean operatoin over standardized y1 * y2
+        **kwargs: passed to final numpy.mean operation over standardized y1 * y2
 
     Returns: correlation array
     """
+
     y1 = (y1 - y1.mean(axis=axis, keepdims=True)) / (y1.std(axis=axis, keepdims=True, ddof=0) + eps)
     y2 = (y2 - y2.mean(axis=axis, keepdims=True)) / (y2.std(axis=axis, keepdims=True, ddof=0) + eps)
     return (y1 * y2).mean(axis=axis, **kwargs)
+
+
+def oracle_corr_corrected(repeated_outputs: ArrayLike) -> np.ndarray:
+    """
+    Compute the corrected oracle correlations per neuron.
+
+    Args:
+        repeated_outputs (array-like): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+
+    Returns:
+        array: Corrected oracle correlations per neuron
+    """
+
+    var_noise, var_mean = [], []
+    for repeat in repeated_outputs:
+        var_noise.append(repeat.var(axis=0))
+        var_mean.append(repeat.mean(axis=0))
+    var_noise = np.mean(np.array(var_noise), axis=0)
+    var_mean = np.var(np.array(var_mean), axis=0)
+    return var_mean / np.sqrt(var_mean * (var_mean + var_noise))
+
+
+def oracle_corr(repeated_outputs: ArrayLike) -> np.ndarray:
+    """
+    Compute the oracle correlations per neuron.
+
+    Args:
+        repeated_outputs (array-like): numpy array with shape (images, repeats, responses), or a list of lists of
+            responses (with len: num of repeats) per image.
+
+    Returns:
+        array: Oracle correlations per neuron
+    """
+
+    if len(repeated_outputs.shape) == 3:
```

Comment:
Also this line would fail `repeated_outputs` is really a list of list so this should be changed. I suggest you do a check against the actual type of the object instead.